File: main/variable.cpp
#include <iostream>

using namespace std;

class Variable {
public:
    Variable();
    Variable(string s);
private:
    string name;
};

Variable::Variable() {
    name = "";
}

Variable::Variable(string s) {
    name = s;
}


int main() {
    cout << "test" << endl;
    return 0;
};

File: kot/Variable.kt
package kot

public class Variable(var name: String = "", var coef: Int = 1) {
    init {
        printInfo()
    }

    fun printInfo() {
        println(name)
        println(coef)
        println(setCoef(name))
    }
    
    fun setCoef(input: String): Int {
        val regex = Regex("""^\d+""")
        val matchResult = regex.find(input)
        return matchResult?.value?.toIntOrNull() ?: 1
    }
}
